src/diffusers/models/modeling_utils.py

```diff
@@ -1482,18 +1481,21 @@ def _load_pretrained_model(
                     " for them. Alternatively, make sure you have `safetensors` installed if the model you are using"
                     " offers the weights in this format."
                 )
-            if offload_folder is not None:
+            else:
                 os.makedirs(offload_folder, exist_ok=True)
             if offload_state_dict is None:
                 offload_state_dict = True
 
+        # Caching allocator warmup
```

Comment:
Expanding the comment to discuss why we only warm up `if device_map is not None` would be helpful.